src/docs/customize/component-customization.mdx

```diff
@@ -0,0 +1,184 @@
+import TextField from '../../lib/components/ui/TextField';
+import { RUIProvider } from '../../lib/provider';
+
+React UI often provides more variants of components, so a user can
+pick the most suitable variant.
+
+There two ways to configure component:
+
+1. Pass props directly to a specific component.
+2. Wrap application (or its part) with `RUIProvider` and pass arbitrary props for
+a specific component through `globalProps` prop.
+
+### Basic Usage
+
+To define global props, you need to import `RUIProvider` first:
+
+```js
+import { RUIProvider } from '@react-ui-org/react-ui';
+```
+
+Then wrap application (or its part) with `RUIProvider` with `globalProps` prop object,
+where its keys are names of components and its values are objects with arbitrary
+props you want to pass to a specific component.
+
+Keys conform to actual names of components:
+
+```jsx
+<RUIProvider globalProps={{
+  Button: { priority: 'filled' },
+  TextField: { variant: 'filled' },
+  SelectField: { variant: 'filled' },
+}}>
+  //...
+</RUIProvider>
+```
+
+See working example:
+
+<Playground>
+  {() => {
+    const [variant, setVariant] = React.useState('filled');
+    return (
+      <RUIProvider
+        globalProps={{
+          Button: { priority: variant },
+          TextField: { variant },
+          SelectField: { variant },
+        }}
+      >
+        <Toolbar align="bottom">
+          <ToolbarItem>
+            <SelectField
+              changeHandler={(e) => setVariant(e.target.value)}
+              id="variant"
+              label="Select variant of Select Field"
+              options={[
+                {
+                  label: 'filled',
+                  value: 'filled',
+                },
+                {
+                  label: 'outline',
+                  value: 'outline',
+                },
+              ]}
+              value={variant}
+            />
+          </ToolbarItem>
+          <ToolbarItem>
+            <TextField
+              id="my-text-field"
+              label="Text Field"
+            />
+          </ToolbarItem>
+          <ToolbarItem>
+            <Button
+              clickHandler={() => {}}
+              id="my-button"
+              label="Button"
+            />
+          </ToolbarItem>
+        </Toolbar>
+      </RUIProvider>
+    );
+  }}
+</Playground>
+
+### Best practices
+
+Global props is a powerful tool that allows you to pass arbitrary props
+to every component you want. It has been made to simplify repeatedly
+set props that are common for your whole project.
+
+For that reason **we do recommend to set only global props which have
+impact on appearance**. Do not use it for props that are used to be
+unique, e.g `clickHandler`,`closeHandler`, `disabled`, `id`, `label`,  etc.
+
+## Translations
+
+Some components may contain tooltips which improve components' accessibility.
+All texts are in English by default and can be translated to other languages.
+
+### Translation Approaches
```

Comment:
I would consider moving translation documentation to a separate page. Its close relatation to globalProps is in my view only an implementation detail.

Reply:
Yes, @bedrich-schindler please create a separate page for this and link the pages through.